### PR TITLE
Refac rolling

### PR DESCRIPTION
**Pull request by MaxBenChrist**

### CODE REVIEW

**Comment by MaxBenChrist** [comment]
I refactored the documentation / docstring for the rolling mechanism, I also sneaked in a ISSUE_TEMPLATE for github and a notebooks-requirements.txt

**Comment by nils-braun** [comment]
Hm, I love it if you put many issues in one PR ;-P

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Oh no, you encountered a problem while using *tsfesh*.
+
+We, the maintainers, are happy to help you. When opening an issue, please provide the following information to us:
+
+1. Your operating system
+2. The version of *tsfresh* that you are using
+3. The data on which the problem occurred (please do not upload 1000s of time series but try to boil the problem down to a singular one)
+4. A minimal code snippet which produces the problem/bug
```
Comment (nils-braun) [inline .github/ISSUE_TEMPLATE.md]: Reproduces

File: docs/text/faq.rst [hunk 2]
```diff
@@ -2,13 +2,15 @@ FAQ
...
 
-    2. *Is it possible to extract features from rolling/shifted time series?*
-       Yes, there is the option `rolling` for the :func:`tsfresh.feature_extraction.extract_features` function.
-       Set it to a non-zero value to enable rolling. In the moment, this just rolls the input data into
-       as many time series as there are time steps - so there is no internal optimization for rolling calculations.
-       Please see :ref:`rolling-label` for more information.
+    2. **Is it possible to extract features from rolling/shifted time series?**
+       Yes, the :func:`tsfresh.dataframe_functions.roll_time_series` function allows to conviniently create a rolled
+       time series datframe from your data. You just have to transform your data into one of the supported tsfresh
+       formats.
```
Comment (nils-braun) [inline docs/text/faq.rst]: You could link to the formats description

File: docs/text/rolling.rst [hunk 3]
```diff
@@ -3,26 +3,28 @@
...
-Imagine the following situation: you have the data of EEG measurements, that
-you want to use to classify patients into healthy and not healthy (we oversimplify the problem here).
-You have e.g. 100 time steps of data, so you can extract features that may forecast the healthiness
-of the patients. But what would happen if you had only the recorded measurement for 50 time steps?
-The patients would be as healthy as with 100 time steps. So you can easily increase the amount of
-training data by reusing time series cut into smaller pieces.
-
-Another example is streaming data, e.g. in Industry 4.0 applications. Here you typically get one
-new data row at a time and use this to predict machine failures for example. To train you model,
+The "time" column of a time series container gives a sequential order to the data.
```
Comment (nils-braun) [inline docs/text/rolling.rst]: Maybe state, that time is not only temporal data

File: docs/text/rolling.rst [hunk 4]
```diff
@@ -3,26 +3,28 @@
...
+Imagine the following situation:
+You have the data of certain sensors (e.g. EEG measurements) as the base to classify patients into healthy and not
+healthy (we oversimplify the problem here).
+Lets say you have data of 100 time steps of data, so you can extract features that may forecast the healthiness
+of the patients.
+If you also have target values for 100 time steps, then you could predict the healthiness of the patient in every
+time step.
+But then, you want to extract features for time series in every time step.
+The rolling mechanism creates such time series for every time step by adding for example the values of the sensors of
+the last 10 time steps.
```
Comment (nils-braun) [inline docs/text/rolling.rst]: No I do not understand what you want to say in this paragraph

File: docs/text/rolling.rst [hunk 5]
```diff
@@ -3,26 +3,28 @@
...
+
+Another example can be found in streaming data, e.g. in Industry 4.0 applications.
+Here you typically get one new data row at a time and use this to predict machine failures for example. To train you model,
```
Comment (nils-braun) [inline docs/text/rolling.rst]: Typo with full stop

File: docs/text/rolling.rst [hunk 6]
```diff
@@ -50,9 +51,11 @@ but rolling works on all 3 data formats :ref:`data-formats-label`):
 | 2  | t9   | 11 | 13 |
 +----+------+----+----+
 
-where you have measured two values (x and y) for two different entities (1 and 2) in 4 or 2 time steps.
+where you have measured the values from two sensors x and y for two different entities (id 1 and 2) in 4 or 2 time
+steps (t1 to t9).
 
-If you set `rolling` to 0, the feature extraction works on
+Now, we can use :func:`tsfresh.utilities.dataframe_functions.roll_time_series` to get consecutive sub-time series.
+E.g. if you set `rolling` to 0, the feature extraction works on
```
Comment (nils-braun) [inline docs/text/rolling.rst]: Rolling = 0 is not possible with this function. It is just using the function not at all.

File: tsfresh/utilities/dataframe_functions.py [hunk 7]
```diff
@@ -327,12 +327,20 @@ def normalize_input_to_internal_representation(df_or_dict, column_id, column_sor
...
-    (time is here the abstract sort order defined by the sort column). For each rolling step a new id will be
-    created, with the name "id={id}, shift={shift}" where the id is the former id of the column and shift is the
-    amount of "time" shifts. ATTENTION: This will (obviously) create new IDs! The sign of rolling defines the
-    direction of time rolling.
-    For more information, please see :ref:`rolling-label`.
+    Roll the (sorted) data frames for each kind and each id separately in the "time" domain
+    (which is represented by the sort order of the sort column given by `column_sort`).
+
+    For each rolling step, a new id with the scheme "id={id}, shift={shift}" where the id is the former id of the
+    column and shift is the amount of "time" shifts.
```
Comment (nils-braun) [inline tsfresh/utilities/dataframe_functions.py]: Grammar

File: tsfresh/utilities/dataframe_functions.py [hunk 8]
```diff
@@ -327,12 +327,20 @@ def normalize_input_to_internal_representation(df_or_dict, column_id, column_sor
...
+
+    A few remarks:
+
+     * This method will create new IDs!
+     * The sign of rolling defines the direction of time rolling, a positive value means we are going back in time
+     * It is possible to shift time series of different lenghts but
+     * We assume that the time series are uniformly sampled
+     * The shift parameter should not be larger than the longest length of the individual time series
```
Comment (nils-braun) [inline tsfresh/utilities/dataframe_functions.py]: Only the sign matters! We o not use the absolute value!

**Pull request merged**